app/src/main/java/me/hufman/androidautoidrive/phoneui/viewmodels/CarDrivingStatsModel.kt

```diff
@@ -1,252 +1,454 @@
+		it.tryAsJsonPrimitive("ecoRangeWon")?.tryAsDouble
+	}.combine(units) { value, units ->
+		units.distanceUnits.fromCarUnit(value)
+	}.format("%.1f").addUnit(unitsDistanceLabel)
+
+	fun update() {
+		// any other updates
+		_idriveVersion.value = carInfo.capabilities["hmi.version"]
+	}
+
+	/* JEZIKK additions */
+	val engineTemp = carInfo.cachedCdsData.liveData[CDS.ENGINE.TEMPERATURE].map {
+		it.tryAsJsonObject("temperature")?.tryAsJsonPrimitive("engine")?.tryAsDouble
+	}.format("%.0f").addUnit(unitsTemperatureLabel)
+	val oilTemp = carInfo.cachedCdsData.liveData[CDS.ENGINE.TEMPERATURE].map {
+		it.tryAsJsonObject("temperature")?.tryAsJsonPrimitive("oil")?.tryAsDouble
+	}.format("%.0f").addUnit(unitsTemperatureLabel)
+	val speedActual = carInfo.cdsData.liveData[CDS.DRIVING.SPEEDACTUAL].map {
+		it.tryAsJsonPrimitive("speedActual")?.tryAsDouble
+	}.format("%.0f") //.addUnit(unitsAverageSpeedLabel)
+	val speedDisplayed = carInfo.cdsData.liveData[CDS.DRIVING.SPEEDDISPLAYED].map {
+		it.tryAsJsonPrimitive("speedDisplayed")?.tryAsDouble
+	}.format("%.0f").addUnit(unitsAverageSpeedLabel)
+	val tempInterior = carInfo.cdsData.liveData[CDS.SENSORS.TEMPERATUREINTERIOR].map {
+		it.tryAsJsonPrimitive("temperatureInterior")?.tryAsDouble
+	}.format("%.1f").addUnit(unitsTemperatureLabel)
+	val tempExterior = carInfo.cdsData.liveData[CDS.SENSORS.TEMPERATUREEXTERIOR].map {
+		it.tryAsJsonPrimitive("temperatureExterior")?.tryAsDouble
+	}.format("%.1f").addUnit(unitsTemperatureLabel)
+
+	/*
+		0 - Initialisierung -> initialization
+		1 - Tractionmodus -> Traction mode
+		2 - Komfortmodus -> Comfort mode
+		3 - Basismodus -> Basic mode
+		4 - Sportmodus -> Sports mode
+		5 - Sportmodusplus -> Sport+
+		6 - Racemodus -> Race
+		7 - Ecopro ->EcoPro
+		8 - Ecoproplus -> EcoPro+
+		9 - Komfortmoduserweitert -> Comfort mode extended
+		xx - Adaptive Mode
+		15 - Unknown
+
+		Some cars have "+" modes (sport+, comfort+) and some other the same modes are called "Individual".
+	 */
+	//val drivingMode = carInfo.cdsData.liveData[CDS.DRIVING.MODE].map {
+	val drivingMode = carInfo.cachedCdsData.liveData[CDS.DRIVING.MODE].map {
+		val a = it.tryAsJsonPrimitive("mode")?.tryAsInt
+		if (a != null){
+			if(a == 2)
+				"Comfort"
+			else if (a == 9)
+				"Comfort+"
+			else if (a == 4)
+				"Sport"
+			else if (a== 5)
+				"Sport+"
+			else if(a==6)
+				"Race"
+			else if(a==7)
+				"EcoPro"
+			else if(a==8)
+				"EcoPro+"
+			else
+				"-"
+		}
+		else
+			"-"
+		//"DEBUG: $a"
+
+	}
+	/*
+		2 - handbrake
+		0 - Bremsensymbol_AUS -> Brake symbol off
+		1 - Bremsensymbol_Gelb -> yellow
+		2 - Bremsensymbol_Rot -> red
+		3 - Bremsemsymbol_Gruen -> green
+		4 - PARK_Gelb
+		8 - PARK_Rot
+		12 - Park_Gruen
+		16 - AutoP_Gelb
+		32 - AutoP_Rot
+		48 - AutoP_Gruen
+		127 - Unknown
+	 */
+
+	val parkingBrake = carInfo.cachedCdsData.liveData[CDS.DRIVING.PARKINGBRAKE].map {
+		val pB = it.tryAsJsonPrimitive("parkingBrake")?.tryAsInt
+		if (pB != null && pB == 2) {

```

Comment:
No need to check for `!= null`, you can check the value directly: `if (pb == 2) {`

Reply:
done